sdks/go/examples/native_wordcap/nativepubsubio/subscriptiontracker.go

```diff
@@ -0,0 +1,80 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nativepubsubio
+
+import "math"
+
+// The SubscriptionRTracker maintains a single entry string to keep up with
+// the PubSub subscription being used in the NativeRead SDF.
+type SubscriptionRTracker struct {
+	Subscription string
+	Done         bool
+}
+
+// NewSubscriptionRTracker returns an RTracker wrapping the
+// provided subscription and a "Done" boolean.
+func NewSubscriptionRTracker(entry string) *SubscriptionRTracker {
+	return &SubscriptionRTracker{Subscription: entry, Done: false}
+}
+
+// TryClaim returns true iff the given position is a string and matches the underlying
+// subscription ID.
+func (s *SubscriptionRTracker) TryClaim(pos interface{}) bool {
+	posString, ok := pos.(string)
+	return ok && posString == s.Subscription
+}
+
+// TrySplit is a no-op for the StaticRTracker in the normal case and moves the subscription
+// to the residual in the checkpointing case, marking itself as done to keep the logical checks
+// around SDF data loss happy.
+func (s *SubscriptionRTracker) TrySplit(frac float64) (primary, residual interface{}, err error) {
+	if math.Abs(frac-0.000) < 0.0001 {
```

Comment:
Does `frac == 0.0` not work? If so, that's unfortunate - I imagine this won't be a totally uncommon pattern.

Reply:
It's likely overkill in this case but I usually err on the side of not fully trusting floating point numbers.

Reply:
Definitely more of me covering my bases out of habit than necessity for how checkpointing is implemented

Reply:
I'd vote we verify `== 0.0` works and then use it - people will copy the patterns they see here, and this is unnecessarily complex if 0.0 does work. Plus this theoretically could be bad for sufficiently large restrictions (if you truly got a restriction of 0.00001 or something).

Regardless, the math.Abs is definitely unnecessary, any negative value should equate to a 0.0 split

Reply:
Validated that it did work with thee exact fractional check, should be good to go now

Reply:
Did you forget to push the change?

Reply:
Oops

Reply:
Made the commit and missed the most important part